src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -384,6 +433,22 @@ private void ProcessWrite(byte[] buffer, int offset, int count, LazyAsyncResult
             }
         }
 
+        private AsyncProtocolRequest InitiateWrite(byte[] buffer, int offset, int count, LazyAsyncResult asyncResult)
+        {
+            _sslState.CheckThrow(authSuccessCheck: true, shutdownCheck: true);
+            ValidateParameters(buffer, offset, count);
+
+            if (Interlocked.Exchange(ref _nestedWrite, 1) == 1)
+            {
+                throw new NotSupportedException(SR.Format(SR.net_io_invalidnestedcall, "Write", "write"));
+            }
+
+            // If this is an async operation, get the AsyncProtocolRequest to use.
+            // We do this only after we verify we're the sole write operation in flight.
+            AsyncProtocolRequest asyncRequest = GetOrCreateProtocolRequest(ref _writeProtocolRequest, asyncResult);
+            return asyncRequest;
```

Comment:
Actually, why is this code even part of InitiateWrite?  InitiateWrite is used in two places: from ProcessWrite (part of Begin/EndWrite) and from WriteAsyncImpl... but WriteAsyncImpl just ignores this return value.  It would be better to just make InitiateWrite return void and leave this GetOrCreateProtocolRequest code in ProcessWrite, where it's actually needed.